projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py

```diff
@@ -122,6 +122,48 @@ def forward(self, a):
 def ViewDynamicExpandModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(2, 4, 30, 384))
 
+# ==============================================================================
+#
+class CollapseStaticModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([2,3,4], torch.float32, True)])
+
+    def forward(self, a):
+        return torch.ops.prims.collapse(a, 1, 2)
+
+
+@register_test_case(
+    module_factory=lambda: CollapseStaticModule())
+def CollapseStaticModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(2,3,4))
+
+# ==============================================================================
+#
+class CollapseDynamicModule(torch.nn.Module):
```

Comment:
I imagine the lowering works for fully dynamic inputs too, right? If so, we should have a test case that is fully dynamic (this one collapses static with dynamic). Also, can we test the case where the collapse range encompasses more than two elements?

Reply:
Added these 2 tests. I also added a test where the input is rank-1, which revealed a bug (basically upstream MLIR doesn't allow collapse_shape if nothing collapses) Always a good idea asking for extra tests :-D

Reply:
Nice!